### ISSUE TITLE
The input line disappears at startup if the toolbar is hidden

### ISSUE THREAD

**Issue by kkebo** [holzschu/a-shell#865]
## Steps to reproduce

1. Run `hideToolbar`
1. Quit all a-Shell windows
1. Open a-Shell

## Expected behavior

![IMG_0537](https://github.com/user-attachments/assets/48c8734f-fc57-4728-89d7-419824297797)

## Actual behavior

The input line disappears from the screen every time I launch a-Shell. If I scroll the screen once with a trackpad, it will appear.

https://github.com/user-attachments/assets/2d23bbe5-3286-4fd1-92cb-504b0bc898bf

If the toolbar is not hidden, this issue doesn't occur.

## Environment

- [.bashrc](https://github.com/kkebo/dotfiles/blob/d487b79dd0f03061cc5cc58236c4c92d364a98a6/.bashrc.ashell)
- [.profile](https://github.com/kkebo/dotfiles/blob/d487b79dd0f03061cc5cc58236c4c92d364a98a6/.profile.ashell)
- a-Shell 1.15.8 (437) (App Store)
- iPadOS 18.2 Developer Beta 4
- iPad Pro 13-inch (M4)
- [Magic Keyboard](https://www.apple.com/shop/product/MWR53LL/A/magic-keyboard-for-ipad-pro-13%E2%80%91inch-m4-us-english-black)

**Comment by holzschu** [comment]
I'm unable to reproduce that one with my current test machine (which is on iOS 16). There is one point where I scroll the webView to some place that is computed (in order to make it scroll to the bottom). That may have been causing the system to scroll to a negative value, I added a `max(value, 0)`  to avoid that (but again, no way to test if that fixes things).

**Comment by kkebo** [comment]
Thank you for your investigation. I'll try to build a-Shell again, and test on simulators and real devices if possible.

By the way, here is the Stage Manager + Slo-mo video. It's easier to see how the cursor moved.

https://github.com/user-attachments/assets/158ddfcd-1e95-48f8-9621-101b9e611fa6

**Comment by kkebo** [comment]
I have successfully built the wasm3 branch of a-Shell, and I reproduced the issue with it too.

And then, I've confirmed that the issue would be solved by modifying [this line](https://github.com/holzschu/a-shell/blob/628fc1cc70652a590342405c132bbd37b25ef8bf/a-Shell/SceneDelegate.swift#L3104):

```diff
diff --git a/a-Shell/SceneDelegate.swift b/a-Shell/SceneDelegate.swift
index 082b1aa3..185664fe 100644
--- a/a-Shell/SceneDelegate.swift
+++ b/a-Shell/SceneDelegate.swift
@@ -3101,7 +3101,7 @@ class SceneDelegate: UIViewController, UIWindowSceneDelegate, WKNavigationDelega
                     // reset, so that we don't send it twice:
                     windowPrintedContent = ""
                     // scroll to the bottom of the webview: https://stackoverflow.com/questions/51659208/how-to-programmatically-scroll-ios-wkwebview-swift-4
-                    let scrollPoint = CGPoint(x: 0, y: webView!.scrollView.contentSize.height - webView!.frame.size.height)
+                    let scrollPoint = CGPoint(x: 0, y: max(webView!.scrollView.contentSize.height - webView!.frame.size.height, 0))
                     webView?.scrollView.setContentOffset(scrollPoint, animated: true)
                 } else {
                     NSLog("commandRunning= \(currentCommand)")
```

Before:

https://github.com/user-attachments/assets/541a73f0-5883-4a63-a282-3d1adddecc1d

After:

https://github.com/user-attachments/assets/f2990b8f-b87d-42da-b9ab-02c13beb60ca

**Comment by kkebo** [comment]
On my device:

- `webView!.scrollView.contentSize.height`: 1151.0
- `webView!.frame.size.height`: 1176.0
- `webView!.scrollView.contentSize.height - webView!.frame.size.height`: -25.0

**Comment by kkebo** [comment]
The issue has been fixed in 1.15.9 (443). Thank you!
